### PR TITLE
Contract calldata api

### PR DESCRIPTION
**Pull request by kyledewy**
Hey I wanted to use the library to construct calldata for contract functions, but I wasn't able to find anything.

I just did a quick change to see if I could expose the api for this, but I'm happy to clean it up if it makes sense.

### CODE REVIEW

File: src/contract/mod.rs [hunk 1]
```diff
@@ -105,37 +105,35 @@ impl<T: Transport> Contract<T> {
...
             .and_then(|function| function.encode_input(&params.into_tokens()))
-            .map(move |data| {
-                let Options {
-                    gas,
-                    gas_price,
-                    value,
-                    nonce,
-                    condition,
-                } = options;
+            .unwrap()
```
Comment (tomusdrw) [inline src/contract/mod.rs:116]: The error is there for a reason, simply `unwrap`ing will cause panics in case the function is not found.

**Comment by tomusdrw** [comment]
Awesome, cheers!

File: src/contract/mod.rs (hunk 1 above)
Comment (kyledewy) [reply src/contract/mod.rs]: Thanks for pointing me towards [#17](https://github.com/tomusdrw/rust-web3/issues/17) .. ethabi_derive looks really nice :ok_hand: 
Sorry about the unwrap() .. should have been an issue not a pull request :roll_eyes:

Comment (tomusdrw) [reply src/contract/mod.rs]: No worries @kyledewy, thanks a lot for the contribution!

**Pull request merged**